### PR TITLE
Only show queue tabs the reviewer has permissions for

### LINKED ISSUE #7640: Only show queue tabs the reviewer has permissions for
The reviewer queue tabnav bar shows tabs that some reviewers don't have access to. The tab bar should be updated to adhere to permissions. See:
https://github.com/mozilla/addons-server/blob/360517b91eaf4311cf3f1c5c972014c9badd550e/src/olympia/reviewers/templatetags/jinja_helpers.py#L68-L115

We should add in checks for legacy reviewer permission, and moderated reviews permissions.

### PR DESCRIPTION
**Pull request by kewisch**
[fixes #7640] 

This gates the tabs per permission. I've tested this locally, although I had to create the moderated reviews group manually. I suspect we have this created on dev/stage/prod and are actually using the Ratings:Moderate permission.

### CODE REVIEW

**Comment by kewisch** [comment]
I've refactored the tests a bit to use the existing helper, I've added a `totalqueues` parameter to that, which will make sure the right number of queues exists for the queue being tested. Does that sound sufficient to you?

File: src/olympia/reviewers/tests/test_views.py [hunk 1]
```diff
@@ -972,13 +972,21 @@ def _test_get_queue(self):
...
-        assert a.attr('href') == self.url
+        links = doc('.tabnav li a')
+        link = links.eq(position)
+
+        assert links.length == totalqueues
+        assert link.text() == '%s (%s)' % (name, count)
+        assert link.attr('href') == self.url
+        if per_page:
+            assert doc('.data-grid-top .num-results').text() == (
+                u'Results %s\u20131 of %s' % per_page)
```
Comment (wagnerand) [inline src/olympia/reviewers/tests/test_views.py]: Is there a parameter missing? You have two substitutes.

Comment (kewisch) [reply src/olympia/reviewers/tests/test_views.py]: What kind of refactoring are you imagining? I guess I could do `(per_page[0], per_page[1])`, or 

```javascript
page_count, page_max = per_page
assert doc('.data-grid-top .num-results').text() == (
    u'Results %s\u20131 of %s' % (page_count, page_max)
```

File: src/olympia/reviewers/tests/test_views.py [hunk 2]
```diff
@@ -972,13 +972,21 @@ def _test_get_queue(self):
         for addon in self.expected_addons:
             self.get_queue(addon)
 
-    def _test_queue_count(self, position, name, count):
-        response = self.client.get(self.url)
+    def _test_queue_count(self, name, position, count, totalqueues,
```
[the remark this answers is not in the document]
Comment (kewisch) [reply src/olympia/reviewers/tests/test_views.py]: How about `_test_queue_count_size` ? It is kind of still about count, just different kinds of counts :)

**Comment by kewisch** [comment]
> Please also add tests that test the permissions specifically, which is what this patch is about.

This is implicitly tested given that each of the _test_queue_count calls is for a different type of queue, and given it searches for the names as well it will check if the queue is present based on the permission requested for the class. I'd say this is sufficient, but if you prefer I am happy to add a more explicit test.

File: src/olympia/reviewers/tests/test_views.py (hunk 1 above)
Comment (wagnerand) [reply src/olympia/reviewers/tests/test_views.py]: Yes, having two separate arguments might be easier for other developers using that function.

File: src/olympia/reviewers/tests/test_views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wagnerand) [reply src/olympia/reviewers/tests/test_views.py]: Or maybe `_test_queue_count_size_layout`. It also contains tests for the order and such.

**Comment by wagnerand** [comment]
I would appreciate if you could add more explicit tests. It's not obvious that the current tests cover this. Also, some don't. For example `TestModeratedQueue::test_queue_count` (without change in https://github.com/mozilla/addons-server/pull/7642/files#diff-fe0574eafc8df03690d218dbde4694a2R1493) passes even with the original code.

File: src/olympia/reviewers/tests/test_views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kewisch) [reply src/olympia/reviewers/tests/test_views.py]: Going with `_test_queue_layout` and making the parameters a bit more descriptive. I don't want to make this super long, and size/count also contributes to layout.

File: src/olympia/reviewers/tests/test_views.py (hunk 1 above)
Comment (kewisch) [reply src/olympia/reviewers/tests/test_views.py]: Turns out I actually only need one of those arguments.

**Pull request merged**